Utils/Extensions/StringExtensions.swift

```diff
@@ -52,6 +52,31 @@ public extension String {
         return str as! String
     }
 
+    /**
+    Truncates a String in the middle only if it's longer than `maxLength`
+
+      "ABCDEF".truncate(4)
+      // "AB…EF"
+
+    :param: maxLength The maximum length of the String.
+
+    :returns: A truncated String
+    */
+    func truncate(#maxLength: Int) -> String {
+        if count(self) >= maxLength {
+            let index1 = advance(self.startIndex, maxLength / 2)
+            var truncatedString = self.substringToIndex(index1)
+
+            truncatedString += "..."
```

Comment:
In my testing this does not insert a newline. What I suspect you're seeing is the label wrapping, and this is a convenient place to put it. Can you post a screenshot of an ellipsized dialog with this using a real ellipsis character?